sdk/client-core/core/src/main/java/io/clientcore/core/implementation/http/rest/RestProxyImpl.java

```diff
@@ -54,12 +53,11 @@ Response<?> send(HttpRequest request) {
 
     @SuppressWarnings({"try", "unused"})
     @Override
-    public Object invoke(Object proxy, Method method, Consumer<HttpRequest> requestCallback,
-                         SwaggerMethodParser methodParser, HttpRequest request) {
+    public Object invoke(Object proxy, SwaggerMethodParser methodParser, HttpRequest request) {
         // If there is 'RequestOptions' apply its request callback operations before validating the body.
         // This is because the callbacks may mutate the request body.
-        if (requestCallback != null) {
-            requestCallback.accept(request);
+        if (request.getRequestOptions() != null) {
+            request.getRequestOptions().getRequestCallback().accept(request);
```

Comment:
Could request callback be null here?

Reply:
No. `RequestOptions` is always created with an instance of an empty callback and users cannot set their own callback. We add things to it by using the `Consumer.andThen()` method in our own `addHeader()`, `setBody()`, etc.